src/Simulation/Common/QubitManager.cs

```diff
@@ -127,7 +127,7 @@ protected IEnumerable<Qubit> QubitsInUse(StackFrame frame)
                 return Qubit.NO_QUBITS;
             }
 
-            return frame.Locals.Concat(frame.QubitsInArgument ?? Array.Empty<Qubit>())
+            return frame.Locals.Keys.Concat(frame.QubitsInArgument ?? Array.Empty<Qubit>())
```

Comment:
Are you sure the change works here. Before it was returning duplicates and now it does not.

Reply:
All uses of this methods call Distinct on the returned IEnumerable so the change is correct, but I should probably clean it up to remove the necessity to call Distinct altogether. Thanks for pointing this out!